clippy_lints/src/unit_types/unit_cmp.rs

```diff
@@ -16,10 +16,13 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>) {
                 sym::assert_ne_macro | sym::debug_assert_ne_macro => "fail",
                 _ => return,
             };
-            let Some((left, _, _)) = find_assert_eq_args(cx, expr, macro_call.expn) else {
+            let Some((lhs, rhs, _)) = find_assert_eq_args(cx, expr, macro_call.expn) else {
                 return;
             };
-            if !cx.typeck_results().expr_ty(left).is_unit() {
+            if !cx.typeck_results().expr_ty(lhs).is_unit() {
+                return;
+            }
+            if is_unit_expr(lhs) || is_unit_expr(rhs) {
```

Comment:
Please switch those tests, `is_unit_expr()` is cheaper than `expr_ty()`, so `is_unit_expr()` should go first.